trl/trainer/dpo_trainer.py

```diff
@@ -652,6 +655,48 @@ def process_row(features, processing_class, max_prompt_length, max_completion_le
 
         return output
 
+    def _prepare_fsdp(self, model: PreTrainedModelWrapper):
+        # Adpated from accelerate: https://github.com/huggingface/accelerate/blob/739b135f8367becb67ffaada12fe76e3aa60fefd/src/accelerate/accelerator.py#L1421
+        from torch.distributed.fsdp.fully_sharded_data_parallel import FullyShardedDataParallel as FSDP
+
+        # Check if the model is already a FSDP model due to `Manual Wrapping` and if so,
+        # don't wrap it again
+        if not isinstance(model, FSDP):
+            self.accelerator.state.fsdp_plugin.set_auto_wrap_policy(model)
+            fsdp_plugin = self.accelerator.state.fsdp_plugin
+            kwargs = {
+                "sharding_strategy": fsdp_plugin.sharding_strategy,
+                "cpu_offload": fsdp_plugin.cpu_offload,
+                "auto_wrap_policy": fsdp_plugin.auto_wrap_policy,
+                "mixed_precision": fsdp_plugin.mixed_precision_policy,
+                "sync_module_states": fsdp_plugin.sync_module_states,
+                "backward_prefetch": fsdp_plugin.backward_prefetch,
+                "forward_prefetch": fsdp_plugin.forward_prefetch,
+                "use_orig_params": fsdp_plugin.use_orig_params,
+                "param_init_fn": fsdp_plugin.param_init_fn,
+                "ignored_modules": fsdp_plugin.ignored_modules,
+                "limit_all_gathers": fsdp_plugin.limit_all_gathers,
+                "device_id": self.accelerator.device,
+            }
+            model = FSDP(model, **kwargs)
+            if fsdp_plugin.activation_checkpointing:
```

Comment:
If I'm not mistaken, we don't need activation checkpointing since we never do a backward pass with the reference model. I think this block can thus be removed